### ISSUE TITLE
Potential bug in CLS compliance checker in the presence of explicit interface member implementations

### ISSUE THREAD

**Issue by GeeLaw** [dotnet/roslyn#52134]
**Version Used**: Hmmm not sure... I'm looking at this from Visual Studio 2019 (`VisualStudio.16.Release/16.9.2+31112.23`) with C# Tools (`3.9.0-6.21160.10+59eedc33d35754759994155ea2f4e1012a9951e3`), targeting `netcoreapp3.1` (shouldn't make a difference).

**Steps to Reproduce**:

```csharp
[assembly: CLSCompliant(true)]

public interface IFoo
{
  void Bar();
  void Baz(out int value);
}

public class Foo : IFoo
{
  public void Bar() { }
  void IFoo.Bar() { }
  // CS3006: Overloaded method differing only in ref and out, or in array rank, is not CLS-compliant.
  public void Baz(ref int value) { value = 1; }
  void IFoo.Baz(out int value) { value = 0; }
}
```

**Expected Behavior**: No compiler warning.

**Actual Behavior**: Compiler warning CS3006.

**Reasoning.** Since `IFoo.Baz` is explicitly implemented, there's no risk of confusing it with `Foo.Baz`. CLS compliance rules also do not mention `ref` and `out` (I thought that it would even be invalid in CTS without EIMI). Moreover, `IFoo.Bar` and `Foo.Bar` are considered to be CLS-compliant, yet they (vacuously) differ only in `ref` and `out`, or in array rank.
